src/Language/PureScript/Make.hs

```diff
@@ -384,3 +389,34 @@ buildMakeActions outputDir filePathMap foreigns usePrefix =
 
   progress :: ProgressMessage -> Make ()
   progress = liftIO . putStrLn . renderProgressMessage
+
+-- | For a given CoreFn module and corresponding FFI module file path,
+-- check that there is exactly one FFI declaration for each
+checkForeignDecls :: CF.Module CF.Ann -> FilePath -> SupplyT Make ()
+checkForeignDecls m path = do
+  jsStr <- lift $ makeIO (internalError . show) (readFile path)
+  js <- either throwUnableToParse pure $ JS.parse jsStr path
+
+  foreignIdents <- fmap S.fromList . lift . Make . lift $
+    catchE (getExportedIdentifiers (runModuleName mname) js)
+           (internalError . show)
+  let importedIdents = S.fromList $ map (runIdent . fst) (CF.moduleForeign m)
+
+  let extraneousFFI = foreignIdents  S.\\ importedIdents
+  let missingFFI    = importedIdents S.\\ foreignIdents
```

Comment:
It may well be possible to do this more neatly, without traversing both sets twice. I'm not aware of any function to achieve this, though.